onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -15,23 +15,65 @@ namespace onnxruntime {
    etc.
    */
 
-// ComputeOffset: compute offset into a tensor. This is essentially the dot-product of
-// index and stride, restricted to the specified number of axes.
-static inline size_t ComputeOffset(const std::vector<int64_t>& index, const std::vector<size_t>& stride, int64_t num_axes) {
-  size_t offset = 0;
-  for (int64_t j = 0; j < num_axes; ++j) {
-    offset += index[j] * stride[j];
+typedef struct MultiIndex {
+  size_t index;
+  size_t upper_bound;
+  int64_t stride;
+  MultiIndex() {}
```

Comment:
In the default constructor, you leave all the variables uninitialised.

Reply:
Yes, to save time. The default constructor is called when creating an array. Even if they were initialized, they would be overwritten after. This index is not exposed in the header, I don't expect this index to be used somewhere else in the code.

Reply:
Then you should have an init function instead of using operator= to fill it?

Reply:
done

Reply:
nit: the 'typedef' isn't needed.

also a comment capturing why the values are not explicitly initialized would be helpful to prevent someone innocently changing that in the future causing a perf regression.

Reply:
> > I am good with this change. Probably needs sign-off from @skottmckay.
> 
> Was just waiting to see if there were any perf issues from the ORT_ENFORCE in the loop. For this release I'm not convinced we need that given the issue wasn't the calculations picking a bad offset, so if there's an issue we can back that out.

I could not measure the differnce between with/without ORT_ENFORCE in the loops. It was not significant so I let them.

Reply:
OK. I would prefer for DoTransposeEltWise to not be public given this bounds check, but that can be a separate PR.